lib/matrix_check_sum.rb

```diff
@@ -3,8 +3,40 @@
 # of numbers in row i is the same as the sum of numbers in column i for i = 0 to row.length-1
 # If this is the case, return true. Otherwise, return false.
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) - n is the number of elements in the matrix
```

Comment:
Take a look at your loops, you have nested loops and each loop runs n number of times so n x n = n^2.